pallet.el

```diff
@@ -187,10 +184,29 @@ use `pallet--package-archives-copy' if USE-COPY is true."
   "Create a Caskfile dependency set from PACKAGE-LIST."
   (let ((depends-list '()))
     (dolist (package package-list)
-      (push (format "(depends-on \"%s\")" package) depends-list))
+      (push (pallet--format-dependency package) depends-list))
     (let ((depends-list (sort depends-list #'string<)))
       (mapconcat 'identity depends-list "\n"))))
 
+(defun pallet--format-dependency (package)
+  "Return a string correctly formatting a dependency PACKAGE."
+  (let ((depend-args (list (symbol-name (cask-dependency-name package)))))
+    (-when-let (version (cask-dependency-version package))
+      (nconc depend-args (list version)))
+    (-when-let (fetcher (cask-dependency-fetcher package))
+      (nconc depend-args (list fetcher))
+      (let ((url (cask-dependency-url package)))
+        (nconc depend-args (list url)))
+      (-when-let (ref (cask-dependency-ref package))
+        (nconc depend-args (list :ref ref)))
+      (-when-let (branch (cask-dependency-branch package))
+        (nconc depend-args (list :branch branch)))
+      (-when-let (files (cask-dependency-files package))
+        (nconc depend-args (list :files files))))
```

Comment:
OK -- I think quotes and backpacks are probably more idiomatic, but for my money `list` is cleaner.